### ISSUE TITLE
Demeter chain fails to return self

### ISSUE THREAD

**Issue by ing200086** [mockery/mockery#816]
When trying to have a demetered chain which changes a value object, it would be good to have the mocked object return itself. Currently this does not seem to work, and the returned object is a Mockery_1_demeter_** class.

Using the example code provided, I would like to be able to perform an action like this.

```
$mock = \Mockery::mock('CaptainsConsole');
$mock->shouldReceive('foo->bar->zebra->alpha->selfDestruct')->andReturnSelf();
```

**Comment by robertbasic** [comment]
@ing200086 can't you achieve that by just doing

`$mock = $mock->shouldReceive('foo->bar->baz')->andReturn($mock)`?

**Comment by robertbasic** [comment]
Yeah, this is not possible at the moment, disregard my previous comment.

**Comment by robertbasic** [comment]
This feature gets implemented with #848 

Given the following setup:

``` php
<?php
class CaptainsConsole
{
    public function foo(): self
    {
        return $this;
    }

    public function bar(): self
    {
        return $this;
    }

    public function zebra(): self
    {
        return $this;
    }

    public function alpha(): self
    {
        return $this;
    }

    public function selfDestruct(): self
    {
        return $this;
    }
}
```

This mock:

``` php
$mock = \Mockery::mock('CaptainsConsole');
$mock->shouldReceive('foo->bar->zebra->alpha->selfDestruct')->andReturnSelf();
```

will work as expected and return an instance of `CaptainsConsole`.

**Comment by robertbasic** [comment]
#848 got merged, this got implemented as well.
